pkg/controllers/data.go

```diff
@@ -8,85 +8,88 @@ import (
 
 	. "github.com/CESARBR/knot-cloud-storage/pkg/entities"
 	"github.com/CESARBR/knot-cloud-storage/pkg/interactor"
+	"github.com/CESARBR/knot-cloud-storage/pkg/logging"
 	"github.com/gorilla/mux"
 )
 
 const maxItemsAllowedToRequest = 100
 
 type DataController struct {
 	DataInteractor *interactor.DataInteractor
+	logger         logging.Logger
 }
 
 type errorMessage struct {
 	error   bool
 	message string
 }
 
-func NewDataController(dataInteractor *interactor.DataInteractor) *DataController {
-	return &DataController{dataInteractor}
+func NewDataController(dataInteractor *interactor.DataInteractor, logger logging.Logger) *DataController {
+	return &DataController{dataInteractor, logger}
 }
 
-func respondWithError(w http.ResponseWriter, code int, msg string) {
-	respondWithJson(w, code, map[string]string{"message": msg})
+func (d *DataController) respondWithError(w http.ResponseWriter, code int, msg string) {
+	d.respondWithJson(w, code, map[string]string{"message": msg})
 }
 
-func respondWithJson(w http.ResponseWriter, code int, payload interface{}) {
+func (d *DataController) respondWithJson(w http.ResponseWriter, code int, payload interface{}) {
 	response, _ := json.Marshal(payload)
 	w.Header().Set("Content-Type", "application/json")
 	w.WriteHeader(code)
-	w.Write(response)
+	_, err := w.Write(response)
+	if err != nil {
+		d.logger.Error(err)
+	}
```

Comment:
Shouldn't the error be returned? :thinking: 
Same comment as [here](https://github.com/CESARBR/knot-cloud-storage/pull/16#discussion_r407674116).

Reply:
The main `handler` function is called by the `mux` router and it doesn't require an error to be returned, as we can see [here](https://www.gorillatoolkit.org/pkg/mux#Router.HandleFunc).

Reply:
Done!